partiql-tests-data/fail/static-analysis/primitives/timestamp-constructor.ion

```diff
@@ -0,0 +1,655 @@
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid Day field - minus before day",
+  statement: "TIMESTAMP '2021-03--10T00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid Day field - plus before day",
+  statement: "TIMESTAMP '2021-03-+10T00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid literal Format - random string",
+  statement: "TIMESTAMP 'timestamp'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid HOUR field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01T24:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid HOUR field - Negative value",
+  statement: "TIMESTAMP '2021-03-01T-01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid HOUR field - plus sign",
+  statement: "TIMESTAMP '2021-03-01T+01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid HOUR field - plus sign",
+  statement: "TIMESTAMP '2021-03-01T+01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid MINUTE field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01T00:60:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid MINUTE field - negative value",
+  statement: "TIMESTAMP '2021-03-01T00:-01:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid MINUTE field - plus sign",
+  statement: "TIMESTAMP '2021-03-01T00:+01:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+// Setting to 62 to take account of leap second, if any implement does care.
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid SECOND field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01T00:00:62'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid SECOND field - negative value",
+  statement: "TIMESTAMP '2021-03-01T00:00:-01'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid SECOND field - plus sign",
+  statement: "TIMESTAMP '2021-03-01T00:00:+01'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid Precision - negative value",
+  statement: "TIMESTAMP(-1) '2021-03-01T00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+// TIMESTAMP - WITH TIME ZONE - SQL FORMAT
```

Comment:
nit: could be helpful to quote/cite what the SQL spec says about time zone bounds as they might not be as intuitive as the other timestamp fields

Reply:
This is actually interesting. The SQL spec says time zone range is between -14:00 to +14:00. However, Ion Spec says that the time zone range should be -23:59 to +23:59. I do believe that PartiQL, being a super set of both Ion and PartiQL, should support time zone range of -23:59 to +23:59 as time zone range. 

That say, the word **format**, is chosen because of the lack of better word. It is used mostly for distinguish the usage of `T` vs `space` to separate data and time component, and the usage of `+00:00` and `Z` to indicate `zulu timezone`. 

Consider `TIMESTAMP '2023-01-01 00:00:00Z'` 
- It uses space to separate date and time, but Z to indicate zulu timezone.  
- This is a combination of "SQL format" and "RFC 3339" format. May be leave it to the implementation to decide whether it wants to support this format? 

Consider `TIMESTAMP '2023-01-01 00:00:00+23:59`
- This timestamp has a time zone that exceeds the SQL specified time zone range, but we should probably just say we allow this. 
- In my (poorly chosen) terminology, this is SQL format, but this does not indicate it follows SQL semantics for time zone....

Reply:
Thanks for the info. PartiQL being a super set of both Ion and SQL could be an important point to document somewhere in the tests related to the time zone range, perhaps when we get to adding the evaluation tests.